src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -581,20 +593,36 @@ module.exports = function<T, P, I, TI, C, CX>(
     }
 
     switch (workInProgress.tag) {
-      case IndeterminateComponent:
-        return mountIndeterminateComponent(current, workInProgress, priorityLevel);
-      case FunctionalComponent:
-        return updateFunctionalComponent(current, workInProgress);
-      case ClassComponent:
-        return updateClassComponent(current, workInProgress, priorityLevel);
+      case IndeterminateComponent: {
+        const next = mountIndeterminateComponent(current, workInProgress, priorityLevel);
+        return next;
+      }
+      case FunctionalComponent: {
+        const next = updateFunctionalComponent(current, workInProgress);
+        memoizeProps(workInProgress);
+        return next;
+      }
+      case ClassComponent: {
+        const next = updateClassComponent(current, workInProgress, priorityLevel);
+        const instance = workInProgress.stateNode;
+        workInProgress.memoizedState = instance.state;
+        workInProgress.memoizedProps = instance.props;
+        workInProgress.pendingProps = null;
```

Comment:
We had trouble keeping things in sync when we split these. It might be helpful to move this inside of `updateClassComponent` and similarly for the others so that it is easier to tell when this needs to happen relative to other things. I'm not 100% sure this should definitely be at the very end in all cases.